endpoints-control/src/main/java/com/google/api/control/Client.java

```diff
@@ -420,27 +409,24 @@ public void run() {
 
   private void flushAndScheduleQuota() {
     if (resetIfStopped()) {
-      log.log(Level.FINE, "did not schedule quota flush: client is stopped");
+      log.atFine().log("did not schedule quota flush: client is stopped");
       return;
     }
     int interval = quotaAggregator.getFlushIntervalMillis();
     if (interval < 0) {
-      log.log(Level.FINE, "did not schedule quota flush: caching is disabled");
+      log.atFine().log("did not schedule quota flush: caching is disabled");
       return; // cache is disabled, so no flushing it
     }
 
     if (isRunningSchedulerDirectly()) {
-      log.log(Level.FINE, "did not schedule check flush: no scheduler thread is running");
+      log.atFine().log("did not schedule check flush: no scheduler thread is running");
       return;
     }
 
-    log.log(Level.FINE, "flushing the quota aggregator");
+    log.atFine().log("flushing the quota aggregator");
     Stopwatch w = Stopwatch.createUnstarted(ticker);
     List<AllocateQuotaRequest> reqs = quotaAggregator.flush();
-    if (log.isLoggable(Level.FINE)) {
-      log.log(Level.FINE,
-          String.format("flushing %d quota from the quota aggregator", reqs.size()));
-    }
+    log.atFine().log("flushing %d quota from the quota aggregator", reqs.size());
```

Comment:
If `reqs.size()` is time-consuming to calculate, perhaps [this work should be avoided](https://google.github.io/flogger/best_practice#avoid-work).

Reply:
It's not for a list.